generic_scripts/nlk.py

```diff
@@ -0,0 +1,303 @@
+        fieldy = self.calc_magnetic_field(r)[1]
+        return x_pos, fieldy
+
+    def get_horizontal_magnetic_field(self, x0=0):
+        """NLK horizontal field at a vertical plane x=x0 for y ∈ [-12, 12] mm.
+
+        Parameters
+        ----------
+        x0 : float, optional
+            x coordinate of the horizontal plane.
+
+        Returns
+        -------
+        y_pos: numpy.array
+            Values of y where the field was computed.
+
+        fieldy: numpy.array
+            Horizontal field in [T] at the plane x=x0.
+        """
+        y_pos = _np.linspace(-12, 12)*1e-3
+        x_pos = _np.ones(y_pos.shape) * x0
+        r = _np.vstack([x_pos, y_pos])
+        fieldy = self.calc_magnetic_field(r)[0]
+        return y_pos, fieldy
+
+    @staticmethod
+    def si_nlk_kick(scale=1, fit_monomials=None, plot_flag=False, r0=0.0):
+        """Generates the NLK polynom_b, its horizontal kick and the integrated
+        field at y=0. Useful for set the NLK pulse in the Sirius model.
+
+        Parameters
+        ----------
+        scale : float, optional
+            Multiplier of the NLK max field, by default 1.
+
+        fit_monomials : array like or list, optional
+            The multipoles indices or degrees of the terms included in the
+            polynom_b, by default is numpy.arange(10, dtype=int)
+
+        plot_flag : bool, optional
+            If True, plots the field profile, by default False.
+
+        r0 : float, optional
+            Closed orbit x coordinate, by default 0.0.
+
+        Returns
+        -------
+        x : numpy.array
+            Values of x where the field was computed.
+
+        integ_field : numpy.array
+            Integrated NLK field in [T.m].
+
+        kickx : numpy.array
+            Horizontal kick in [rad].
+
+        polynom_b : numpy array of shape (fit_monomials,)
+            Polynom B that describes the NLK kick.
+        """
+
+        nlk_length = 0.45
+        if fit_monomials is None:
+            fit_monomials = _np.arange(10, dtype=int)
+        else:
+            fit_monomials = _np.asarray(fit_monomials)
+
+        nlk = NLK()
+        x, mag_field = nlk.get_vertical_magnetic_field()
+
+        brho, *_ = beam_rigidity(energy=3)  # energy in [GeV]
+        integ_field = scale * mag_field * nlk_length  # [T.m]
+        kickx = integ_field / brho  # [rad]
+
+        coeffs = _polyfit(x=x-r0, y=kickx, deg=fit_monomials)
+        fit_kickx = _polyval(x=x-r0, c=coeffs)
+        if plot_flag:
+            _plt.figure()
+            _plt.scatter(1e3*(x-r0), 1e3*kickx, label="data points")
+            _plt.plot(
+                1e3*(x-r0), 1e3*fit_kickx, c=[0, 0.6, 0], label="fitted curve")
+            _plt.xlabel('X [mm]')
+            _plt.ylabel("Kick @ 3 GeV [mrad]")
+            _plt.title("NLK Profile")
+            _plt.legend()
+
+        polynom_b = _np.zeros(1 + fit_monomials.max())
+        polynom_b[fit_monomials] = -coeffs/nlk_length
```

Comment:
I think this is a bug...
You need to do:
```
polynom_b[fit_monomials] = -coeffs[fit_monomials]/nlk_length
```
don't you?
Again, it is weird that this didn't raised an error in your tests... Am I missing something?

Reply:
Yes, its is really a bug. Solved, thanks!